### PR TITLE
Barebones Client UI class for Blak chat app

### PR DESCRIPTION
**Pull request by p0lygun**
Adds barebones UI class to develop UI further
files  

image

`app` dir is where all code will go  
`app/lib` is where the main libraries of the app will be  
`app/utils` is for helper functions and helper data  
`app/utils/server.py` has a simple WebSocket server to help develop frontend API  

`app/lib/kivy_manager` is the main file that starts the client UI and connects to the server 
`app/__main__.py` is the entry point of our app so that we can run it via
`poetry run -m app` (when in client dir)

@noahlias pls read this and ask questions if any 
**and pls submit a review only after reading the code**

### CODE REVIEW

File: client/app/lib/kivy_manager.py [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+
+import websockets
+from kivy.app import App
+from kivy.logger import Logger
+
+
+class ClientUI(App):
+    """Main Class to Build frontend on."""
+
+    other_task = None
```
Comment (noahlias) [inline client/app/lib/kivy_manager.py]: need to add variable type hit and var unused

File: client/app/utils/server.py [hunk 2]
```diff
@@ -0,0 +1,23 @@
+"""This is dummy server that is used when developing New API for client."""
+
+import asyncio
+import random
+
+import websockets
+from loguru import logger
```
Comment (noahlias) [inline client/app/utils/server.py]: we should to add locally websocket server log to store it

File: client/pyproject.toml [hunk 3]
```diff
@@ -7,6 +7,9 @@ license = "MIT"
 
 [tool.poetry.dependencies]
 python = "3.10.*"
+Kivy = "^2.1.0"
```
Comment (noahlias) [inline client/pyproject.toml:10]: sorry for the dependencies, we need to change tool poetry name
```suggestion
[tool.poetry]
name = "Blak"
version = "0.1.0"
description = "itsn't a bug its a feature"
authors = ["Author 1 <Gorgeous-Ghouls@github.com>"]
license = "MIT"
```

File: client/app/utils/server.py [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+async def echo(websocket):
+    """Send Random message back."""
+    async for message in websocket:
+        logger.debug(f"Received {message}")
+        await websocket.send(str(random.random()))
+
+
+async def main():
+    """Create and Host the server"""
+    async with websockets.serve(echo, "localhost", 8765):
```
Comment (noahlias) [inline client/app/utils/server.py:19]: maybe this we can set it  to server part, it more like task need to be done in backend

**Review by noahlias — CHANGES_REQUESTED** [review]
Some advice from me

File: client/app/lib/kivy_manager.py (hunk 1 above)
Comment (p0lygun) [reply client/app/lib/kivy_manager.py]: Yh need to remove that

File: client/app/utils/server.py (hunk 2 above)
Comment (p0lygun) [reply client/app/utils/server.py]: I'm not sure i understand, we will use kivy's logger so loguru will be removed

File: client/pyproject.toml (hunk 3 above)
Comment (p0lygun) [reply client/pyproject.toml:10]: I don't understand it

File: client/app/utils/server.py (hunk 4 above)
Comment (p0lygun) [reply client/app/utils/server.py:19]: This is just a dummy server we use to make frontend API

File: client/pyproject.toml (hunk 3 above)
Comment (p0lygun) [reply client/pyproject.toml:10]: Ok i get it, will do

**Review by p0lygun — COMMENTED** [review]
@noahlias re-review pls

**Review by noahlias — APPROVED** [review]

**Pull request merged**